app/src/main/java/com/example/tasky/feature_agenda/domain/use_case/GetReminder.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.tasky.feature_agenda.domain.use_case
+
+import com.example.tasky.feature_agenda.domain.model.AgendaItem
+import com.example.tasky.feature_agenda.domain.repository.ReminderRepository
+import com.example.tasky.util.Resource
+import com.example.tasky.util.Result
+
+class GetReminder(
+    private val repository: ReminderRepository
+) {
+    suspend operator fun invoke(reminderId: String): Result<AgendaItem.Reminder> {
+        val result = repository.getReminder(reminderId)
+
+        return when(result) {
+            is Resource.Error -> Result.Error(result.message ?: "Unknown Error")
+            is Resource.Success -> Result.Success(result.data)
+            else -> Result.Error(result.message ?: "Unknown Error")
+        }
+    }
+}
```

Comment:
I'm personally not a big fan of such empty use cases. If you do this for the whole project, you'll end up with 15-20 of such use cases (seems like it's already the case haha) which do nothing else than calling a repository's function. Clean architecture is fine and all, but I personally think we need to adjust it to our project while keeping the benefits. I've seen so many projects where I had to click through 5+ classes and layers just to see what a function call does. That doesn't make code more readable, but much harder to understand.

Instead, you could just call the repo's function in your ViewModel which is totally fine, since the repo's abstraction sits in domain.